listenbrainz/webserver/static/js/src/recent/RecentListens.tsx

```diff
@@ -0,0 +1,306 @@
+
+                    const isListenReviewable =
+                      Boolean(recordingMBID) ||
+                      artistMBIDs?.length ||
+                      Boolean(trackMBID) ||
+                      Boolean(releaseGroupMBID);
+                    /* eslint-disable react/jsx-no-bind */
+                    const additionalMenuItems = (
+                      <>
+                        <ListenControl
+                          title="Pin this recording"
+                          icon={faThumbtack}
+                          action={this.updateRecordingToPin.bind(this, listen)}
+                          dataToggle="modal"
+                          dataTarget="#PinRecordingModal"
+                        />
+                        {isListenReviewable && (
+                          <ListenControl
+                            title="Write a review"
+                            icon={faPencilAlt}
+                            action={this.updateRecordingToReview.bind(
+                              this,
+                              listen
+                            )}
+                            dataToggle="modal"
+                            dataTarget="#CBReviewModal"
+                          />
+                        )}
+                      </>
+                    );
+                    /* eslint-enable react/jsx-no-bind */
+                    return (
+                      <ListenCard
+                        key={`${listen.listened_at}-${listen.track_metadata?.track_name}-${listen.track_metadata?.additional_info?.recording_msid}-${listen.user_name}`}
+                        showTimestamp
+                        showUsername
+                        listen={listen}
+                        currentFeedback={this.getFeedbackForRecordingMsid(
+                          listen.track_metadata?.additional_info?.recording_msid
+                        )}
+                        updateFeedbackCallback={this.updateFeedback}
+                        newAlert={newAlert}
+                        additionalMenuItems={additionalMenuItems}
+                      />
+                    );
+                  })}
+                </div>
+                {!listens.length && (
+                  <h5 className="text-center">No listens to show</h5>
+                )}
+              </>
+            )}
+          </div>
+          <div className="col-md-4" />
+          {currentUser && (
+            <>
+              <PinRecordingModal
+                recordingToPin={recordingToPin || listens[0]}
+                newAlert={newAlert}
+                onSuccessfulPin={(pinnedListen) =>
+                  newAlert(
+                    "success",
+                    "",
+                    `Successfully pinned ${pinnedListen.track_metadata.track_name}`
+                  )
+                }
+              />
+              <CBReviewModal
+                listen={recordingToReview || listens[0]}
```

Comment:
`listens[0]` here feels a bit wrong. Maybe the component should handle null/undefined better?

Reply:
I was thinking the same